src/storage/QueryBaseProcessor.inl

```diff
@@ -182,20 +97,215 @@ cpp2::ErrorCode QueryBaseProcessor<REQ, RESP>::checkAndBuildContexts(const REQ&
+            if (!schema) {
+                VLOG(1) << "Cant find edgeType " << edgeContext_.edgeType_;
+                return false;
+            }
+            auto field = schema->field(propName);
+            if (field == nullptr) {
+                VLOG(1) << "Can't find related prop "
+                        << propName << " on edge " << edgeExp->alias();
+                return false;
+            }
+            return true;
+        }
+        case Expression::kVariableProp:
+        case Expression::kDestProp:
+        case Expression::kInputProp:
+            return false;
+        default: {
+            VLOG(1) << "Unsupport expression type! kind = "
+                    << std::to_string(static_cast<uint8_t>(exp->kind()));
+            return false;
+        }
+    }
+}
+
+template<typename REQ, typename RESP>
+void QueryBaseProcessor<REQ, RESP>::collectProps(RowReader* reader,
+                                                 folly::StringPiece key,
+                                                 const std::vector<PropContext>& props,
+                                                 FilterContext* fcontext,
+                                                 Collector* collector) {
+    for (auto& prop : props) {
+        switch (prop.pikType_) {
+            case PropContext::PropInKeyType::NONE:
+                break;
+            case PropContext::PropInKeyType::SRC:
+                VLOG(3) << "collect _src, value = " << NebulaKeyUtils::getSrcId(key);
+                collector->collectInt64(NebulaKeyUtils::getSrcId(key), prop);
+                continue;
+            case PropContext::PropInKeyType::DST:
+                VLOG(3) << "collect _dst, value = " << NebulaKeyUtils::getDstId(key);
+                collector->collectInt64(NebulaKeyUtils::getDstId(key), prop);
+                continue;
+            case PropContext::PropInKeyType::TYPE:
+                VLOG(3) << "collect _type, value = " << NebulaKeyUtils::getEdgeType(key);
+                collector->collectInt64(static_cast<int64_t>(NebulaKeyUtils::getEdgeType(key)),
+                                        prop);
+                continue;
+            case PropContext::PropInKeyType::RANK:
+                VLOG(3) << "collect _rank, value = " << NebulaKeyUtils::getRank(key);
+                collector->collectInt64(NebulaKeyUtils::getRank(key), prop);
+                continue;
+        }
+        if (reader != nullptr) {
+            const auto& name = prop.prop_.get_name();
+            auto res = RowReader::getProp(reader, name);
+            if (!ok(res)) {
+                VLOG(1) << "Skip the bad value for prop " << name;
+                continue;
+            }
+            auto&& v = value(std::move(res));
+            if (prop.fromTagFilter()) {
+                fcontext->tagFilters_.emplace(prop.tagFilterKey(), v);
+            }
+            if (prop.returned_) {
+                switch (v.which()) {
+                    case 0:
```

Comment:
maybe use type micro to replace 0,1,2,3 is more readable.

Reply:
Good suggestion